<!doctype html>
<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <title>Genomics-workshop2016 by mspopgen</title>

    <link rel="stylesheet" href="stylesheets/styles.css">
    <link rel="stylesheet" href="stylesheets/github-light.css">
    <meta name="viewport" content="width=device-width">
    <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
  </head>
  <body>
    <div class="wrapper">
      <header>
        <h1>Genomics-workshop2016</h1>
        <p></p>

        <p class="view"><a href="https://github.com/mspopgen/genomics-workshop2016">View the Project on GitHub <small>mspopgen/genomics-workshop2016</small></a></p>


        <ul>
          <li><a href="https://github.com/mspopgen/genomics-workshop2016/zipball/master">Download <strong>ZIP File</strong></a></li>
          <li><a href="https://github.com/mspopgen/genomics-workshop2016/tarball/master">Download <strong>TAR Ball</strong></a></li>
          <li><a href="https://github.com/mspopgen/genomics-workshop2016">View On <strong>GitHub</strong></a></li>
        </ul>
      </header>
      <section>
        <p><em>Population Genomics Workshop 2016, University of Sheffield</em></p>

<h1>
<a id="genome-wide-association-studies-in-structured-populations" class="anchor" href="#genome-wide-association-studies-in-structured-populations" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Genome-wide association studies in structured populations</h1>

<h4>
<a id="michael-stocks" class="anchor" href="#michael-stocks" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Michael Stocks</h4>

<p>By the end of this tutorial you should be able to:</p>

<ul>
<li>Perform a genome-wide association study</li>
<li>Add covariates to the model</li>
<li>Account for multiple testing</li>
<li>Correct for population stratification</li>
</ul>

<h2>
<a id="getting-started" class="anchor" href="#getting-started" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Getting started</h2>

<p>If you type <code>ls</code> and enter in the terminal then you should see two files. <code>gen_RUFF_qc.raw</code> was generated from <code>VCF</code> format using <code>plink</code> and contains the genotypic information for each marker and individual. <code>phe_RUFF.txt</code> is a tab-delimited file containing the phenotypes for each individual.</p>

<p>Type <code>R</code> and enter to begin an <code>R</code> session. You can exit at any time by typing <code>q()</code> to get back to the <code>unix</code> environment.</p>

<h2>
<a id="importing-and-exploring-the-data" class="anchor" href="#importing-and-exploring-the-data" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Importing and exploring the data</h2>

<p align="center">
<img src="https://github.com/mspopgen/genomics-workshop2016/blob/master/ruff-sys.png" width="640" align="center">
</p>

<p>See <a href="http://www.nature.com/ng/journal/v48/n1/full/ng.3443.html"><em>KÃ¼pper et al. (2016)</em></a>, and it's sister publication <a href="http://www.nature.com/ng/journal/v48/n1/full/ng.3430.html"><em>Lamichhaney et al. (2016)</em></a>, for more details.</p>

<div class="highlight highlight-source-r"><pre>library(<span class="pl-smi">GenABEL</span>)</pre></div>

<div class="highlight highlight-source-r"><pre><span class="pl-smi">ruff.data</span> <span class="pl-k">&lt;-</span> load.gwaa.data(<span class="pl-v">phe</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>gen_RUFF_qc.raw<span class="pl-pds">"</span></span>, <span class="pl-v">gen</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>phe_RUFF.raw<span class="pl-pds">"</span></span>, <span class="pl-v">force</span> <span class="pl-k">=</span> <span class="pl-c1">T</span>)</pre></div>

<div class="highlight highlight-source-r"><pre>descriptives.trait(<span class="pl-smi">ruff.data</span>)
descriptives.trait(<span class="pl-smi">ruff.data</span>)</pre></div>

<div class="highlight highlight-source-r"><pre><span class="pl-smi">fo.MLR</span> <span class="pl-k">&lt;-</span> mlreg(<span class="pl-smi">fo</span> <span class="pl-k">~</span> <span class="pl-c1">1</span>, <span class="pl-v">data</span> <span class="pl-k">=</span> <span class="pl-smi">ruff.data</span>, <span class="pl-v">trait</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>binomial<span class="pl-pds">"</span></span>)</pre></div>

<div class="highlight highlight-source-r"><pre>summary(<span class="pl-smi">fo.MLR</span>)

summary(<span class="pl-smi">fo.MLR</span>, <span class="pl-v">top</span> <span class="pl-k">=</span> <span class="pl-c1">30</span>)</pre></div>

<div class="highlight highlight-source-r"><pre>sum(<span class="pl-smi">fo.MLR</span>[, <span class="pl-s"><span class="pl-pds">"</span>P1df<span class="pl-pds">"</span></span>] <span class="pl-k">&lt;</span><span class="pl-k">=</span> <span class="pl-c1">0.0001</span>)</pre></div>

<h2>
<a id="references" class="anchor" href="#references" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>References</h2>
      </section>
      <footer>
        <p>This project is maintained by <a href="https://github.com/mspopgen">mspopgen</a></p>
        <p><small>Hosted on GitHub Pages &mdash; Theme by <a href="https://github.com/orderedlist">orderedlist</a></small></p>
      </footer>
    </div>
    <script src="javascripts/scale.fix.js"></script>
    
  </body>
</html>
